dev/conductor/core/lib/src/clean.dart

```diff
@@ -71,6 +70,25 @@ class CleanCommand extends Command<void> {
       }
     }
     stdio.printStatus('Deleting persistent state file ${stateFile.path}...');
+
+    final RunContext context = RunContext(
+      stateFile: stateFile,
+    );
+    context.run();
+  }
+}
+
+/// Context for cleaning up persistent state file.
+///
+/// This is a frontend-agnostic implementation.
+class RunContext {
+  RunContext({
+    required this.stateFile,
+  });
+
+  final File stateFile;
+
+  void run() {
     stateFile.deleteSync();
```

Comment:
nit: while we're at it, can you make this async (`return stateFile.delete();` and return type `Future<void>` and `await`ed at the call site), that way we don't have to refactor and roll when this involves a network call.